### PR TITLE
Chronos: Add example for onnx

### PR DESCRIPTION
**Pull request by liangs6212**

### CODE REVIEW

**Review by TheaperDeng — COMMENTED** [review]
1. use built-in dataset api
2. evaluate/evaluate_with_onnx has the same output
3. predict_with_onnx is faster than predict
4. network traffic use s2s Forecaster while nyc use autotsest

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# Use ONNX to improve the speed of predict inference
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:1]: to speed up inference

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+This example will demonstrate the effect of ONNX for predict on forecast and autotest, which is expected to be ~5X.
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:2]: which is expected... [delete]

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## Prepare the environment
+We recommend you to use Anaconda to prepare the environment, especially if you want to run on a yarn cluster:
+```
+conda create -n env python=3.7 # "zoo" is conda environment name, you can use any name you like.
+conda activate env
+pip install analytics-zoo[automl]
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:9]: pip install --pre --upgrade analytics-zoo[automl]

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+```
+
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:11]: readme use forecaster as example to walk through, no need to introduce orca context

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+* `--num_workers` The number of workers to be used in the cluster. You can change it depending on your own cluster setting. Default to be 2.
+* `--cores` "The number of cpu cores you want to use on each node. Default to be 4.
+
+## Run on yarn cluster for yarn-client mode after pip install 
+```
+python onnx_nyc_taxi.py/onnx_network_traffic.py --cluster_model yarn
+```
+
+## Prepare data
+**autotest**: We are using the `nyc taxi` provided by NAB, from 2014-07-01 to 2015-01-31 taxi fare information For more details, please refer to [here](https://raw.githubusercontent.com/numenta/NAB/v1.0/data/realKnownCause/nyc_taxi.csv)
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:35]: autotest -> autotsest

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+* `--memory` The memory you want to use on each node. Default to be 10g.
+* `--num_workers` The number of workers to be used in the cluster. You can change it depending on your own cluster setting. Default to be 2.
+* `--cores` "The number of cpu cores you want to use on each node. Default to be 4.
+
+## Run on yarn cluster for yarn-client mode after pip install 
+```
+python onnx_nyc_taxi.py/onnx_network_traffic.py --cluster_model yarn
+```
+
+## Prepare data
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:34]: 1. prepare data 2. fit on forecaster 3. inference with onnx

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+```
+
+Finally, call `fit` on Forecaster.
+```python
+# input tuple.
+x_train, y_train = tsdata_train.to_numpy()
+forecaster.fit((x_train, y_train), epochs=args.epochs)
+```
+
+## Reference
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:70]: Reference -> Results

File: pyzoo/zoo/chronos/examples/onnx/onnx_network_traffic.py [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+                        "You can change it depending on your own cluster setting.")
+    parser.add_argument('--memory', type=str, default="10g",
+                        help="The memory you want to use on each node."
+                        "You can change it depending on your own cluster setting.")
+
+    parser.add_argument("--epochs", type=int, default=2,
+                        help="Max number of epochs to train in each trial.")
+
+    args = parser.parse_args()
+    # init_orca_context
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/onnx_network_traffic.py:60]: no need to use orca context in a standalone forecaster workflow

**Comment by liangs6212** [comment]
http://10.112.231.51:18888/job/ZOO-PR-Python-Spark-2.3-py37/3282/
http://10.112.231.51:18888/job/ZOO-PR-Python-ExampleTests-py37/3409/
http://10.112.231.51:18888/job/ZOO-PR-Python-Spark-3.1-py36k216/1887/
http://10.112.231.51:18888/job/ZOO-PR-Validation/6441/

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
+# Use ONNX to speed up inferencing
+This example will demonstrate how to use ONNX to speed up the inferencing(prediction/evalution) on forecasters and `AutoTSEstimator`.
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:2]: and `TSPipeline`

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+minmax = MinMaxScaler()
+for tsdata in [tsdata_train, tsdata_test]:
+    tsdata.gen_dt_feature(one_hot_features=["HOUR", "WEEK"])\
+          .impute("last")\
+          .scale(minmax, fit=tsdata is tsdata_train)\
+          .roll(lookback=100, horizon=10)
+```
+
+## Fit on forecaster/AutoTSEstimator
+Create and fit on the forecaster/AutoTSEstimator. Please refer to [API doc](placeholder) for detail.
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:32]: replace this placeholder to a real url

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+## Inference with onnx
+All methods involve inferencing supports onnx as backend. You can call `predict_with_onnx` or `evaluate_with_onnx` to use it.
+```python
+# forecaster
+x_test, y_test = tsdata_train.to_numpy()
+forecaster.predict_with_onnx(x_test)
+forecaster.evaluate_with_onnx((x_test, y_test))
+
+# AutoTSEstimator
+autotsestimator.predict_with_onnx(tsdata_test)
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:43]: tspipeline.predict_with_onnx

File: pyzoo/zoo/chronos/examples/onnx/README.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+autotsestimator.evaluate_with_onnx(tsdata_test)
```
Comment (TheaperDeng) [inline pyzoo/zoo/chronos/examples/onnx/README.md:44]: tspipeline.evaluate_with_onnx

File: pyzoo/zoo/chronos/examples/onnx/README.md (hunk 9 above)
Comment (TheaperDeng) [reply pyzoo/zoo/chronos/examples/onnx/README.md:2]: + In this example, onnx speed up the inferencing for ~4X

**Comment by TheaperDeng** [comment]
remember to add test scripts

**Comment by liangs6212** [comment]
http://10.112.231.51:18888/job/ZOO-PR-Python-ExampleTests/3806/
http://10.112.231.51:18888/job/ZOO-PR-Validation/6491/
http://10.112.231.51:18888/job/ZOO-PR-Pip-ExampleTests-py36/63/

**Pull request merged**